packages/ember-template-compiler/lib/plugins/transform-action-syntax.js

```diff
@@ -62,6 +67,24 @@ function isAction(node) {
   return node.path.original === 'action';
 }
 
+function ensureTarget(node, builders) {
+  if (!hashPairForKey(node.hash, 'target')) {
+    let thisTarget = builders.pair('target', builders.path('this'));
+    node.hash.pairs.push(thisTarget);
+  }
+}
+
+function hashPairForKey(hash, key) {
+  for (let i = 0; i < hash.pairs.length; i++) {
+    let pair = hash.pairs[i];
+    if (pair.key === key) {
+      return pair;
+    }
+  }
+
+  return false;
+}
```

Comment:
This function is unfortunately repeated. @rwjblue should the util function be DRY somewhere? And if so where?